google/resource_compute_router.go

```diff
@@ -1,77 +1,125 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
 	"log"
-
-	"strings"
+	"reflect"
+	"strconv"
+	"time"
 
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/compute/v1"
-	"google.golang.org/api/googleapi"
+	"github.com/hashicorp/terraform/helper/validation"
+	compute "google.golang.org/api/compute/v1"
 )
 
 func resourceComputeRouter() *schema.Resource {
 	return &schema.Resource{
 		Create: resourceComputeRouterCreate,
 		Read:   resourceComputeRouterRead,
+		Update: resourceComputeRouterUpdate,
 		Delete: resourceComputeRouterDelete,
+
 		Importer: &schema.ResourceImporter{
-			State: resourceComputeRouterImportState,
+			State: resourceComputeRouterImport,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(240 * time.Second),
+			Update: schema.DefaultTimeout(240 * time.Second),
+			Delete: schema.DefaultTimeout(240 * time.Second),
 		},
 
 		Schema: map[string]*schema.Schema{
-			"name": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateGCPName,
 			},
-
-			"network": &schema.Schema{
+			"network": {
 				Type:             schema.TypeString,
 				Required:         true,
 				ForceNew:         true,
 				DiffSuppressFunc: compareSelfLinkOrResourceName,
 			},
-
-			"description": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-				ForceNew: true,
-			},
-
-			"project": &schema.Schema{
+			"description": {
 				Type:     schema.TypeString,
 				Optional: true,
-				Computed: true,
-				ForceNew: true,
 			},
-
-			"region": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
-				Computed: true,
-				ForceNew: true,
-			},
-
-			"bgp": &schema.Schema{
+			"bgp": {
 				Type:     schema.TypeList,
+				Optional: true,
```

Comment:
bgp is required/forcenew.

Reply:
Turns out it's not actually required! I'm not really sure why you wouldn't include it but the API totally lets you create a router with just a name and network (https://cloud.google.com/compute/docs/reference/rest/v1/routers/insert if you're curious)